src/reflect/scala/reflect/internal/Positions.scala

```diff
@@ -139,17 +142,30 @@ trait Positions extends api.Positions { self: SymbolTable =>
             }
           }
         }
-        for (ct <- tree.children flatMap solidDescendants) validate(ct, tree)
+        for (ct <- childSolidDescendants) validate(ct, tree, trace)
       }
     }
 
     if (!isPastTyper)
-      validate(tree, tree)
+      validate(tree, tree, settings.Yposdebug && (settings.verbose || settings.Yrangepos))
   }
 
-  def solidDescendants(tree: Tree): List[Tree] =
-    if (tree.pos.isTransparent) tree.children flatMap solidDescendants
-    else List(tree)
+  final def solidDescendants(tree: Tree): List[Tree] =
+    if (tree.pos.isTransparent) {
+      val children = tree.children
+      if (children isEmpty) Nil else {
+        val result = new ListBuffer[Tree]
+        def solidDescendantsImpl(tree: Tree) : Unit = {
+          if (tree.pos.isTransparent)
+            tree.children foreach solidDescendantsImpl
```

Comment:
Should this be the ```children``` you caputured instead of ```tree.children``` ?

Reply:
its a different tree, so different children